src/backend/dx12/src/device.rs

```diff
@@ -526,38 +575,110 @@ impl Device {
         format: dxgiformat::DXGI_FORMAT,
         range: &image::SubresourceRange,
     ) -> Result<d3d12::D3D12_CPU_DESCRIPTOR_HANDLE, image::ViewError> {
-        let handle = self.srv_pool.lock().unwrap().alloc_handles(1).cpu;
-
-        let dimension = match kind {
-            image::Kind::D1(..) |
-            image::Kind::D1Array(..) => d3d12::D3D12_SRV_DIMENSION_TEXTURE1D,
-            image::Kind::D2(..) |
-            image::Kind::D2Array(..) => d3d12::D3D12_SRV_DIMENSION_TEXTURE2D,
-            image::Kind::D3(..) |
-            image::Kind::Cube(..) |
-            image::Kind::CubeArray(..) => d3d12::D3D12_SRV_DIMENSION_TEXTURE3D,
-        };
+        #![allow(non_snake_case)]
 
         let mut desc = d3d12::D3D12_SHADER_RESOURCE_VIEW_DESC {
             Format: format,
-            ViewDimension: dimension,
+            ViewDimension: 0,
             Shader4ComponentMapping: 0x1688, // TODO: map swizzle
             u: unsafe { mem::zeroed() },
         };
 
+        let MostDetailedMip = range.levels.start as _;
+        let MipLevels = (range.levels.end - range.levels.start) as _;
+        let FirstArraySlice = range.layers.start as _;
+        let ArraySize = (range.layers.end - range.layers.start) as _;
+
         match kind {
+            image::Kind::D1(_) => {
+                assert_eq!(range.layers, 0..1);
```

Comment:
Should switch to the array version in the case of non-zero base layer

Reply:
Same for the other cases.

Reply:
we can't switch to array version since `D1` kind has only a single layer